azure/src/utils/parseAzureResourceId.ts

```diff
@@ -0,0 +1,23 @@
+/*---------------------------------------------------------------------------------------------
+*  Copyright (c) Microsoft Corporation. All rights reserved.
+*  Licensed under the MIT License. See License.txt in the project root for license information.
+*--------------------------------------------------------------------------------------------*/
+
+import * as types from '../../index';
+import { localize } from "../localize";
+
+export function parseAzureResourceId(id: string): types.ParsedAzureResourceId {
+    const matches: RegExpMatchArray | null = id.match(/\/subscriptions\/(.*)\/resourceGroups\/(.*)\/providers\/(.*)\/(.*)/i);
```

Comment:
I think we should try to make this regex at least a little more precise. For example, subscriptions are always guids, so we could use `([a-f0-9-]+)`. Providers always(ish?) have something like `word.word/word`.

I'm sure we also have regexs somewhere for resource group names, though I wouldn't bother with resource names because those rules are [all over the place](https://docs.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules).

Reply:
Making it stricter could cause issues when there are non-english characters in the id. For example, not even the officially documented resource group regex `[-\w\._\(\)]+` supports Chinese characters. However, the portal lets me create resource groups with Chinese characters.

My stricter regex: https://regexr.com/6prg2

Reply:
Yeah, that's fair. I don't want something excessively-strict for this but I can also see risks if the individual regex groups can take slashes (e.g. the original `(.*)` group for regexs matches slashes too). If something had an unexpected number of slashes for whatever reason, things could break down.